frontend/src/utils/htmlUtils.ts

```diff
@@ -33,3 +33,34 @@ export const convertHtmlToPdf = (
     })
   })
 }
+
+export const onHtmlImgsLoaded = (
+  node: HTMLDivElement,
+  callback: () => void,
+  maxTimeToWaitInMs = 1000,
+) => {
+  const images = node.getElementsByTagName('img')
+  if (images.length === 0) {
+    callback()
+    return
+  }
+
+  let loadedImages = 0
+  setTimeout(callback, maxTimeToWaitInMs) // run callback after maxTimeToWaitInMs to avoid issues with corrupt images never loading
+
+  for (let i = 0; i < images.length; i++) {
+    if (images[i].complete) {
+      loadedImages++
+      if (loadedImages === images.length) {
+        callback()
+      }
+    } else {
+      images[i].onload = () => {
+        loadedImages++
+        if (loadedImages === images.length) {
+          callback()
+        }
```

Comment:
relevant to the above comment, wdyt of something like this?

```suggestion
 for (const image of images) {
  image.complete ? loadedImages++ : (image.onload = () => loadedImages++);
  loadedImages === images.length && callback();
}
```

Reply:
Looks clean, but I think the `if (loadedImages === images.length) { callback() }` check needs to happen as part of the `onload` callback for each image so that the check will be triggered, since this isn't an asynchronous function

Reply:
Can think further on how to optimise this~